app/models/tenant.rb

```diff
@@ -0,0 +1,25 @@
+class Tenant < LandlordRecord
+  has_one :tenant_connection, dependent: :destroy
+
+  after_create_commit :create_tenant_connection
+
+  def self.current
+    current_tenant_connection = TenantConnection.current
+
+    Landlord.switch_to_main do
+      tenant.find_by(tenancy_connection: current_tenant_connection)
+    end
+  end
+
+  def create_tenant_connection
+    @tenancy_connection = TenantConnection.create_from!(from_env_name: Rails.env, from_db_name: "primary", schema_search_path: Landlord::SchemaNameGenerator.generate(name), tenant: self)
+    Landlord.create_tenant(@tenancy_connection.config)
+    self.class.fetch_db_connections
+
+    ## Hot restart puma
+    unless Rails.env.test?
+      pid = File.read(".pumactl_pid").to_i
+      Process.kill("SIGUSR2", pid)
+    end
```

Comment:
we should think about this thing.

Reply:
You mean the hot restart? 👍

Reply:
yeah, I mean, not everyone uses puma, not everyone's aware of the pidfile, etc. normally there'a a pidfile within tmp/pids. We should think about that mostly

Reply:
This is specifically design to work with Puma, I think is a good starting point.
It could be extended after to work with other web servers but I think just puma is ok for now.

Around the `pidfile` it could be configurable, we could assume it is in `temp/pids` and allow users to set it in a initialization file otherwise!